lib/src/front_end/piece_factory.dart

```diff
@@ -626,8 +627,9 @@ mixin PieceFactory {
     //     @meta
     //     // Weird place for comment.
     //     int f() {}
-    var leadingComments =
-        pieces.takeCommentsBefore(returnType.firstNonCommentToken);
+    var firstToken = modifiers.firstWhereOrNull((token) => token != null) ??
```

Comment:
[optional] You could use a non-null filter instead of writing it as a callback.

```suggestion
    var firstToken = modifiers.nonNulls.firstOrNull ??
```

Reply:
Well crap. I landed this before I saw your suggestion. Today I learned about `nonNulls`.